gnomad/assessment/sanity_checks.py

```diff
@@ -18,50 +19,57 @@ def generic_field_check(
     If the number of rows that match the `cond_expr` is 0, then the Table passes that check; otherwise, it fails.
 
-    .. note::
-
+    ..note::
+        
         `cond_expr` and `check_description` are opposites and should never be the same.
         E.g., If `cond_expr` filters for instances where the raw AC is less than adj AC,
         then it is checking sites that fail to be the desired condition (`check_description`)
         of having a raw AC greater than or equal to the adj AC.
-
+        
     :param ht: Table containing annotations to be checked.
     :param cond_expr: Logical expression referring to annotations in ht to be checked.
     :param check_description: String describing the condition being checked; is displayed in terminal summary message.
-    :param display_fields: List of names of ht annotations to be displayed in case of failure (for troubleshooting purposes);
-        these fields are also displayed if verbose is True.
-    :param verbose: If True, show top values of annotations being checked, including checks that pass; if False,
-        show only top values of annotations that fail checks.
+    :param display_fields: List of names of ht annotations to be displayed in case of failure (for troubleshooting purposes); these fields are also displayed if verbose is True.
+    :param verbose: If True, show top values of annotations being checked, including checks that pass; if False, show only top values of annotations that fail checks.
     :param show_percent_sites: Show percentage of sites that fail checks. Default is False.
+    :param n_fail: Previously computed number of sites that fail the conditional checks.
+    :param ht_count: Previously computed sum of sites within hail Table. 
     :return: None
     """
-    ht_orig = ht
-    ht = ht.filter(cond_expr)
-    n_fail = ht.count()
+    if show_percent_sites and (ht_count is None):
+        ht_count = ht.count()
+
+    if n_fail is None and cond_expr:
+        n_fail = ht.filter(cond_expr).count()
+
     if n_fail > 0:
         logger.info("Found %d sites that fail %s check:", n_fail, check_description)
         if show_percent_sites:
-            logger.info("Percentage of sites that fail: %f", n_fail / ht_orig.count())
-        ht = ht.flatten()
-        ht.select("locus", "alleles", *display_fields).show()
+            logger.info(
+                "Percentage of sites that fail: %.2f %%", 100 * (n_fail / ht_count)
+            )
+        ht.select(**display_fields).show()
     else:
         logger.info("PASSED %s check", check_description)
         if verbose:
-            ht_orig = ht_orig.flatten()
-            ht_orig.select(*display_fields).show()
+            ht.select(**display_fields).show()
 
 
 def make_filters_sanity_check_expr(
-    ht: hl.Table, extra_filter_checks: Optional[Dict[str, hl.expr.Expression]] = None
+    ht: hl.Table,
+    extra_filter_checks: Optional[Dict[str, hl.expr.Expression]] = None,
+    rf: bool = False,
```

Comment:
Yes --I went back and forth on this and choose the boolean only because I thought it would be more rigid but  I think the string approach is better since it is more flexible. Ill add the suggestion.